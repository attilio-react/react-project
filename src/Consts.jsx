const consts = { 
    SEARCH_BY_TITLE : 0,
    SEARCH_BY_GENRE : 1,

    SORT_BY_RELEASE_DATE : 0,
    SORT_BY_RATING : 1
}

export {consts}
