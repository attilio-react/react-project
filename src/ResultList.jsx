import React from 'react';

import {ResultItem} from './ResultItem.jsx'

class ResultList extends React.PureComponent {
  render () {
    const {results} = this.props
    return <div>
                {results.map((object, i) => <ResultItem item={object} key={i} />)}
           </div>
  }
}


export {ResultList}


