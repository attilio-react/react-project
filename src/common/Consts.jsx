const consts = { 
    SEARCH_SCREEN : 0,
    DETAIL_SCREEN: 1,


    SEARCH_BY_TITLE : 0,
    SEARCH_BY_GENRE : 1,

    SORT_BY_RELEASE_DATE : 0,
    SORT_BY_RATING : 1,


    PARAM_SORT_BY: 'sortBy',
    PARAM_SORT_ORDER: 'sortOrder',
    PARAM_SEARCH: 'search',
    PARAM_SEARCH_BY: 'searchBy',

    VALUE_BY_GENRE: 'genres',
    VALUE_BY_TITLE: 'title',
    VALUE_BY_RELEASE_DATE: 'release_date',
    VALUE_BY_VOTE_AVERAGE: 'vote_average',

    VALUE_ASC: 'asc',
    VALUE_DESC: 'desc'
}

export {consts}
