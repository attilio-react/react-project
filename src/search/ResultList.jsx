import React from 'react';

import {SearchContext} from './SearchContext.jsx'

import {ResultItem} from 'Common/ResultItem.jsx'

class ResultList extends React.PureComponent {
  render () {
    return (
        <SearchContext.Consumer>
           {ctx => (
           <div>
                {ctx.items.map((object, i) => <ResultItem item={object} clickCallback={ctx.itemClickCb} key={i} />)}
           </div>
           )}
        </SearchContext.Consumer>)
  }
}


export {ResultList}


