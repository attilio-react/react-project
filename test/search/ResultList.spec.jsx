import * as React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {ResultList} from '|search/ResultList.jsx';
import {SearchContext} from '|search/SearchContext.jsx';
import {consts} from '|common/Consts.jsx'

const CONTEXT = {
    itemClickCb: jest.fn(),
    items: [
	{
	    id: 1,
	    poster_path: 'http://hello.com/poster.png',
	    title: 'Movie Title',
	    vote_average: 4.4,
	    release_date: '2011-01-01',
	    runtime: 129,
	    overview: 'This is a movie about ....',
	    genres: ['Horror', 'Thriller']
	},
	{
	    id: 2,
	    poster_path: 'http://hello.com/poster2.png',
	    title: 'Another Movie Title',
	    vote_average: 4.4,
	    release_date: '2012-02-02',
	    runtime: 139,
	    overview: 'This is another movie about ....',
	    genres: ['War', 'History']
	}

    ]
}

describe('ResultList rendering', () => {
  it('Should render', () => {
    const 
        component = renderer.create(
            <SearchContext.Provider value={CONTEXT}>
               <ResultList />
            </SearchContext.Provider>),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('ResultList behaviour', () => {
  it('Should call callback when click on item', () => {
    const 
      resultHeader = mount(
            <SearchContext.Provider value={CONTEXT}>
               <ResultList />
            </SearchContext.Provider>),
      nativeLink = resultHeader.find('a').first();
    nativeLink.simulate('click');
    expect(CONTEXT.itemClickCb).toBeCalledWith(1)
  });

});

