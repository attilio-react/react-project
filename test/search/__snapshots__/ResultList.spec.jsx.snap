// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultList rendering Should render 1`] = `
<div>
  <div
    style={
      Object {
        "border": "1px solid black",
      }
    }
  >
    <a
      onClick={[Function]}
    >
      <p>
        <img
          src="http://hello.com/poster.png"
        />
      </p>
      <p>
        <span>
          2011-01-01
        </span>
         
        <span>
          Horror, Thriller
        </span>
      </p>
      <p>
        <span>
          Movie Title
        </span>
      </p>
    </a>
  </div>
  <div
    style={
      Object {
        "border": "1px solid black",
      }
    }
  >
    <a
      onClick={[Function]}
    >
      <p>
        <img
          src="http://hello.com/poster2.png"
        />
      </p>
      <p>
        <span>
          2012-02-02
        </span>
         
        <span>
          War, History
        </span>
      </p>
      <p>
        <span>
          Another Movie Title
        </span>
      </p>
    </a>
  </div>
</div>
`;
